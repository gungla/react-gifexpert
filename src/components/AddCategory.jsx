import { useState } from "react"

export const AddCategory = ({ onNewCategory }) => {

    const [ inputValue, SetInputValue ] = useState('');

    const onInputChange = ({ target }) => {
        SetInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if( inputValue.trim().length <= 1 ) return;

        //setCategories( categories => [ inputValue, ...categories ]);
        SetInputValue('');
        onNewCategory( inputValue.trim() );
    }

    return (
        <form onSubmit={ onSubmit }>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
    )
}
