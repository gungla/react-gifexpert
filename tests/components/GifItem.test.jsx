import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem />', () => {

    const title = 'Saitama';
    const url = 'https://media0.giphy.com/media/oxbNORcXx76F2/giphy.gif?cid=790b7611efc366d5514dca055f60907589e7a9b4096867c9&rid=giphy.gif&ct=g';

    test('debe de hacer match con el snapshot', () => {
        
        const { container } = render( <GifItem title={title} url={url} /> );        
        expect( container ).toMatchSnapshot();

    });

    test('debe de mostrar la imagen con el URL y al ALT indicado', () => { 

        render( <GifItem title={title} url={url} /> );
        //screen.debug();
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);

     });

     test('debe de mostrar el titulo en el componente', () => { 

        render( <GifItem title={title} url={url} /> );
        expect( screen.getByText( title ) ).toBeTruthy();

      });

})