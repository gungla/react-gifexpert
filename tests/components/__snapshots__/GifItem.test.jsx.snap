// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://media0.giphy.com/media/oxbNORcXx76F2/giphy.gif?cid=790b7611efc366d5514dca055f60907589e7a9b4096867c9&rid=giphy.gif&ct=g"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
